docs/workers.md

```diff
@@ -16,16 +16,36 @@ workers only work with PostgreSQL-based Synapse deployments. SQLite should only
 be used for demo purposes and any admin considering workers should already be
 running PostgreSQL.
 
-## Master/worker communication
+## Main process/worker communication
+
+The processes communicate with each other via a Synapse-specific protocol called
+'replication' (analogous to MySQL- or Postgres-style database replication) which
+feeds streams of newly written data between processes so they can be kept in
+sync with the database state.
+
+Additionally, processes may make HTTP requests to each other. Typically this is
+used for operations which need to wait for a reply - such as sending an event.
+
+As of Synapse v1.13.0, it is possible to configure Synapse to send replication
+via a [Redis pub/sub channel](https://redis.io/topics/pubsub), and is now the
+recommended way of configuring replication. This is an alternative to the old
+direct TCP connections to the main process: rather than all the workers
+connecting to the main process, all the workers and the main process connect to
+Redis, which relays replication commands between processes. This can give a
+significant cpu saving on the main process and will be a prerequisite for
+upcoming performance improvements.
+
+If using Redis the appropriate dependencies must be installed. If using a
+virtual env these can be installed by:
+
+```sh
+pip install matrix-synapse[redis]
+```
+
+Note that these dependencies are included if synapse was installed with `[all]`,
+or if using the debian packages from matrix.org.
```

Comment:
this probably belongs in the "configuration" section (which you might consider renaming to "Setting up workers" or something) rather than here which feels more like an overview of how it works).